cli/ingress-controller/main.go

```diff
@@ -413,10 +414,22 @@ func main() {
 	cacheStores.Plugin = kongPluginInformer.GetStore()
 	informers = append(informers, kongPluginInformer)
 
-	kongClusterPluginInformer := kongInformerFactory.Configuration().V1().KongClusterPlugins().Informer()
-	kongClusterPluginInformer.AddEventHandler(reh)
-	cacheStores.ClusterPlugin = kongClusterPluginInformer.GetStore()
-	informers = append(informers, kongClusterPluginInformer)
+	supportedKongResources, _ := kubeClient.Discovery().ServerResourcesForGroupVersion(
+		configuration.SchemeGroupVersion.String())
+	hasKongClusterPlugin := false
+	for _, resource := range supportedKongResources.APIResources {
+		if resource.Kind == "KongClusterPlugin" {
+			hasKongClusterPlugin = true
+			break
+		}
+	}
```

Comment:
Consider exporting/moving/reusing this function which has identical logic:

https://github.com/Kong/kubernetes-ingress-controller/blob/84ba4a2d23fa9a0bb8bc332230c6793b828dc4c1/internal/ingress/utils/ingressapi.go#L34